### PR TITLE
Fix [replace_requires] for replacements of same reference name

### LINKED ISSUE #17407: [bug] use of replace_requires to update version for the same recipes break in 2.10
### Describe the bug

conan version: 2.10

conan profile:
Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

Build profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

### How to reproduce it

```
conan install --requires=graphene/1.10.8  --build=missing

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
[replace_requires]
glib/2.78.3: glib/2.81.0


======== Computing dependency graph ========
Graph root
    cli
Requirements
    bzip2/1.0.8#d00dac990f08d991998d624be81a9526 - Cache
    glib/2.81.0#70a4237610c8f5e426882afab6d8fd92 - Cache
    graphene/1.10.8#8f0e385bc9e23eba47c2d1697c3f512a - Cache
    libelf/0.8.13#4f70a3555809ae50cc8add44f0f84288 - Cache
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache
    libmount/2.39.2#64850679fa9d5ead878e5076b0cab2a8 - Cache
    libselinux/3.6#5a78ff6ae5034eeaac8da723361a8ce4 - Cache
    pcre2/10.42#ac2227071f79a062167cf2fee7189d66 - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
Build requirements
    autoconf/2.71#f9307992909d7fb3df459340f1932809 - Cache
    flex/2.6.4#e35bc44b3fcbcd661e0af0dc5b5b1ad4 - Cache
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    meson/1.2.2#21b73818ba96d9eea465b310b5bbc993 - Cache
    meson/1.3.1#a17179442f5a99378b58ef0a0f43babc - Cache
    meson/1.3.2#26ce8a76a36cc275cdfee1d757bc6561 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
    pkgconf/2.2.0#6462942a22803086372db44689ba825f - Cache
Replaced requires
    glib/2.78.3: glib/2.81.0
Resolved version ranges
    meson/[>=1.2.3 <2]: meson/1.3.2
    ninja/[>=1.10.2 <2]: ninja/1.12.1
    pkgconf/[>=2.2 <3]: pkgconf/2.2.0
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
ERROR: graphene/1.10.8: Error in validate() method, line 62
	glib_is_shared = self.dependencies["glib"].options.shared
	KeyError: "'glib' not found in the dependency set"

```

It worked in conan 2.9.3

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Bugfix: Fix `[replace_requires]` for replacements of same reference name.
Docs: Omit

Closes https://github.com/conan-io/conan/issues/17407

### CODE REVIEW

File: conans/model/dependencies.py [hunk 1]
```diff
@@ -99,8 +99,12 @@ def from_node(node):
                 added_req.ref = RecipeReference.loads(old_req)
                 d[added_req] = existing
             # Now remove the replaced from dependencies dict
-            for new_req in node.replaced_requires.values():
-                d.pop(new_req, None)
+            for old_req, new_req in node.replaced_requires.items():
+                # But check we're not removing itself if old and new are the same ref
+                # Else we would delete the whole dependency
+                old_ref = RecipeReference.loads(old_req)
```
Comment (AbrilRBS) [inline conans/model/dependencies.py]: Im actually not quite happy with this approach, but it's the best I could find - happy to hear alternatives!

File: test/integration/graph/test_replace_requires.py [hunk 2]
```diff
@@ -178,7 +191,7 @@ class App(ConanFile):
             name = "app"
             version = "0.1"
             settings = "build_type"
-            requires = "zlib/0.1"
+            requires = "zlib/0.2"
```
Comment (AbrilRBS) [inline test/integration/graph/test_replace_requires.py]: To test that we're only checking for name, not full ref

File: conans/model/dependencies.py (hunk 1 above)
Comment (danimtb) [reply conans/model/dependencies.py]: Did you check that `old_req` does not have `old_req.ref.name`?

Comment (AbrilRBS) [reply conans/model/dependencies.py]: Yes, at this point old_req is a string key of the dictonary :)

**Review by czoido — APPROVED** [review]

**Pull request merged**